src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/AbstractSkybox.java

```diff
@@ -148,7 +152,7 @@ public final float getAlpha() {
             }
 
             maxPossibleAlpha *= maxAlpha;
-            if (checkBiomes() && checkHeights() && checkWeather() && checkEffect()) { // check if environment is invalid
+            if (checkBiomes() && checkHeights() && checkWeather() && checkEffect() && checkZRanges() && checkXRanges()) { // check if environment is invalid
```

Comment:
If you could rearrange the checkX and Z functions to be right next to the checkHeights that would be great.
Maybe also renaming checkHeights to be more consistent with these new methods?